base/applications/msconfig/autochkpage.c

```diff
@@ -0,0 +1,230 @@
+
+/* FUNCTIONS *******************************************************************/
+
+INT_PTR CALLBACK
+AutoChkPageWndProc(HWND hDlg, UINT message, WPARAM wParam, LPARAM lParam)
+{
+    HKEY hKey;
+    LONG lResult;
+    WCHAR szValueData[255];
+    DWORD dwcbData = sizeof(szValueData);
+    DWORD dwBufferSize = 0;
+    LPWSTR lpsszMultiString = NULL;
+    size_t szSize;
+    LPWSTR ptr;
+    DWORD dwType;
+
+    UNREFERENCED_PARAMETER(lParam);
+  
+    switch (message)
+    {
+        case WM_INITDIALOG:
+        {
+            hAutoChkDialog = hDlg;
+            SetWindowPos(hDlg, NULL, 10, 32, 0, 0, SWP_NOACTIVATE | SWP_NOOWNERZORDER | SWP_NOSIZE | SWP_NOZORDER);
+
+            // Open the key so we can query it
+            lResult = RegOpenKeyEx(HKEY_LOCAL_MACHINE,
+                                   SubKey,
+                                   0,
+                                   KEY_READ,
+                                   &hKey);
+
+            if (lResult != ERROR_SUCCESS)
+            {
+                // We failed
+                return FALSE;
+            }
+
+            lResult = RegQueryValueExW(hKey,
+                                       ValueName,
+                                       NULL,
+                                       NULL,
+                                       (LPBYTE)szValueData,
+                                       &dwcbData);
+
+            RegCloseKey(hKey);
+
+            if (lResult != ERROR_SUCCESS)
+            {
+                // We failed
+                return FALSE;
+            }
+            else if (lResult == ERROR_SUCCESS)
+            {
+                // Check the array if it's empty
+                if (szValueData[0] == L'\0')
+                {
+                    // If the array is empty, always set the Disable radio button
+                    SendDlgItemMessage(hDlg, IDC_AUTOCHK_DISABLE, BM_SETCHECK, BST_CHECKED, 0);
+                }
+                else
+                {
+                    // Otherwise, always set the Enable radio button
+                    SendDlgItemMessage(hDlg, IDC_AUTOCHK_ENABLE, BM_SETCHECK, BST_CHECKED, 0);
+                }
+            }
+
+            return TRUE;
+        }
+
+        case WM_COMMAND:
+        {
+            switch(LOWORD(wParam))
+            {
+                case IDC_AUTOCHK_ENABLE:
+                {
+                    // Open the key so we can enable autochk
+                    lResult = RegOpenKeyEx(HKEY_LOCAL_MACHINE,
+                                           SubKey,
+                                           0,
+                                           KEY_SET_VALUE,
+                                           &hKey);
+
+                    if (lResult != ERROR_SUCCESS)
+                    {
+                        // We failed
+                        return 1;
+                    }
+
+                    // Set autochk as enabled
+                    lResult = RegSetValueEx(hKey,
```

Comment:
How can I do that? Can you show me an example for this (maybe `wcscat` should do the job)?

Reply:
`wcscat` is on the list of functions that should never be used (`StringCbCatW` would be a more appropriate equivalent).
Since it's a multi-sz value you can just copy the values together, e.g.
```
WCHAR szValueData[255];
PWCHAR pszOldValue;
StringCbCopy(szValueData, sizeof(szValueData), EnableValue);
pszOldValue = szValueData + wcslen(EnableValue) + 1; // points behind the null char
pszOldValue[0] = UNICODE_NULL; // in case there is no old value, terminate
cbData = sizeof(szValueData) - (pszOldValue - szValueData) * sizeof(WCHAR);
lResult = RegQueryValue(... pszOldValue, &cbData ...);
if (lResult != NO_ERROR && lResult != ERROR_FILE_NOT_FOUND) // TODO verify if this is the correct error code
{
    // fail
}

cbData = cbData + (wcslen(EnableValue) + 1) * sizeof(WCHAR);
RegSetValue(...)
```

Reply:
@ThFabba I applied your snippet (with an additional failure check on `StringCbCopy`) however now the code doesn't seem to add `autocheck autochk *` at all...

Here is the Pastebin (code not pushed yet):
https://pastebin.com/0r8XR0xZ

Reply:
StringCbCopy returns 0 on success, not failure. Use the SUCCEEDED or FAILED macro